src/accordion/accordion.ts

```diff
@@ -321,6 +360,34 @@ export class NgbAccordion implements AfterContentChecked {
   private _updateActiveIds() {
     this.activeIds = this.panels.filter(panel => panel.isOpen && !panel.disabled).map(panel => panel.id);
   }
+
+  private _runTransitions() {
+    this._changeDetector.detectChanges();
+    this.panels.forEach(panel => {
+      if (panel.transitionRunning) {
+        const panelElement = this._getPanelElement(panel.id);
+        ngbRunTransition(panelElement !, ngbCollapsingTransition, {
```

Comment:
Could we add a comment explaining why do we restart transitions, even though the transition might already be running?

Reply:
I push again with all the requested changes (I hope I didn't forgot anything !).

Unfortunately, I had to change the accordion code again. This is due to some failing tests on what classes are expected (after the ngbTransition refactor  with the startFn): classes managed by the transitions shouldn't be managed by bindings.

I'm not very happy with the code produced, but this is due to the initial code which would require a BIG refactor, probably in order to avoid ngAfterContentChecked usage (in favor of ngChanges for the accordion, and (change) usage in panels. There are also some inconsistency (in the base code), like:

- `activeIds` documentation : An array or comma separated strings of panel ids that should be opened **initially**,
- `should toggle panels based on "activeIds" values` testing just the opposite.

I did not want to include this refactor, because the review would be much harder.